File: String_manipulation/easy/making_anagrams/making_anagrams.py
def makeAnagram(a, b):
    # Write your code here
    a = sorted(a)
    b = sorted(b)
    # minLength = min(len(a), len(b))
    nbOfDeletion = 0
    counterA = 0
    counterB = 0
    while counterA < len(a) and counterB < len(b):
        if a[counterA] != b[counterB]:
            nbOfDeletion += 1
            if a[counterA] > b[counterB]:
                counterB += 1
            else:
                counterA +=1
        else:
             counterA += 1   
             counterB += 1   
    
    nbOfDeletion += len(a) - counterA
    nbOfDeletion += len(b) - counterB
    return nbOfDeletion

File: String_manipulation/easy/making_anagrams/making_anagrams.js
const fs = require('fs')

function makeAnagram(a, b) {
    // Write your code here
    let alphabetA = Array(26).fill(0);
    let alphabetB = Array(26).fill(0);
    
    for (let i = 0; i < a.length; i++) {
        alphabetA[a.charCodeAt(i) - 97]++;
    }
    
    for (let i = 0; i < b.length; i++) {
        alphabetB[b.charCodeAt(i) - 97]++;
    }
    
    let result = 0;
    for (let i = 0; i < 26; i++) {
        const temp = Math.abs(alphabetA[i] - alphabetB[i]);
        result += temp;
    }
    
    return result
    
}

const a = "abcdef";

const b = "abcde";


const res = makeAnagram(a, b);
console.log(res)
